ecpy_pulses/sequences/views/base_sequence_view.enaml

```diff
@@ -0,0 +1,169 @@
+
+    Conditional: _t_def:
+        condition << item.time_constrained
+
+        Label: def_lab:
+            text = 'Mode'
+        ObjectCombo: def_val:
+            items << list(item.get_member('def_mode').items)
+            selected := item.def_mode
+
+        Splitter: defs:
+            hug_width = 'ignore'
+            SplitItem:
+                Container:
+                    constraints = [hbox(lab_1, val_1),
+                                   align('v_center', lab_1, val_1)]
+                    padding = 0
+                    Label: lab_1:
+                        text << item.def_mode.split('/')[0]
+                    QtLineCompleter: val_1:
+                        hug_width = 'ignore'
+                        text := item.def_1
+                        entries_updater << item.parent.get_bindable_vars
+                        tool_tip = EVALUATER_TOOLTIP
+            SplitItem:
+                Container:
+                    constraints = [hbox(lab_2, val_2),
+                                   align('v_center', lab_2, val_2)]
+                    padding = 0
+                    Label: lab_2:
+                        text << item.def_mode.split('/')[1]
+                    QtLineCompleter: val_2:
+                        hug_width = 'ignore'
+                        text := item.def_2
+                        entries_updater << item.parent.get_bindable_vars
+                        tool_tip = EVALUATER_TOOLTIP
+
+    Notebook: _nb:
+        tabs_closable = False
+
+        Page:
+            title = 'Sequence'
+            FoldableSequenceEditor: editor:
+                sequence := view.item
+
+        Page:
+            title = 'Local vars'
+            DictEditor(FieldFieldEditor): de:
+                de.mapping := item.local_vars
+#                de.operations = ['add','move','remove']
+                #Todo_Restore Add Back autocompletion!
+                
+
+enamldef RootSequenceView(AbstractSequenceView): view:
+    """ View for root sequence.
+
+    """
+    alias additional_pages : include.objects
+
+    alias core : editor.core
+    alias cache : editor.cache
+
+    constraints << [vbox(hbox(dur, *val.items), nb),
+                    align('v_center', val.items[0], val.items[1])] \
+                   if val.items else \
+                   [vbox(dur, nb)]
+
+    CheckBox: dur:
+        text = 'Fix duration'
+        checked := item.time_constrained
+
+    Conditional: val:
+        condition << item.time_constrained
+        Label:
+            text = 'Duration'
+        QtLineCompleter:
+            hug_width = 'ignore'
+            text := item.sequence_duration
+            entries_updater << item.get_bindable_vars
+            tool_tip = EVALUATER_TOOLTIP
+
+    Notebook: nb:
+        tabs_closable = False
+        Page:
+            title = 'Sequence'
+            SequenceEditor: editor:
+                sequence << view.item
+
+        Page:
+            title = 'Context'
```

Comment:
We should make it more obvious that a context is selected. We could update the title for example.